driver/normalizer/normalizer.go

```diff
@@ -15,6 +15,141 @@ var Normalize = Transformers([][]Transformer{
 	{Mappings(Normalizers...)},
 }...)
 
+type opArrHasParams struct {
+	origArr Op
+}
+
+func (op opArrHasParams) Kinds() nodes.Kind {
+	return nodes.KindArray
+}
+
+func (op opArrHasParams) Check(st *State, n nodes.Node) (bool, error) {
+	arr, ok := n.(nodes.Array)
+	if !ok && arr != nil {
+		return false, nil
+	}
+
+    res, err := op.origArr.Check(st, arr)
+    if err != nil {
+    	return false, err
+	}
+	return res, nil
+}
+
+func (op opArrHasParams) Construct(st *State, n nodes.Node) (nodes.Node, error) {
+	arr, err := op.origArr.Construct(st, n)
+	if err != nil || arr == nil {
+		return n, err
+	}
+
+	arr2, ok := arr.(nodes.Array)
+	if !ok && arr2 != nil {
+		return nil, ErrExpectedList.New(arr2)
+	}
+
+	retVal := false
+
+	for _, n := range arr2 {
+		nobj, ok := n.(nodes.Object)
+		if !ok {
+			return nil, ErrExpectedObject.New(nobj)
+		}
+
+		objType, ok := nobj["@type"].(nodes.String)
+		if !ok {
+			return nil, ErrExpectedValue.New(nobj["@type"])
+		}
+
+		if objType == "ParamsKeyword" {
+			retVal = true
+			break
+		}
+	}
+
+	return nodes.Bool(retVal), nil
+}
+
+func funcDefMap(typ string) Mapping {
+	return MapSemantic(typ, uast.FunctionGroup{}, MapObj(
+		Fields{
+			{Name: "Body", Op: Var("body")},
+			{Name: "Identifier", Op: Obj{
+				uast.KeyType: String("IdentifierToken"),
+				uast.KeyPos:      Var("id_fnc_pos"),
+				"LeadingTrivia":  Any(),
+				"TrailingTrivia": Any(),
+				"IsMissing": Bool(false),
+				"Text":      Var("name"),
+				"Value":     Var("name"),
+				"ValueText": Var("name"),
+			}},
+			{Name: "ParameterList", Op: Obj{
+				uast.KeyType: String("ParameterList"),
+				uast.KeyPos: Any(),
+				"OpenParenToken": Any(),
+				"CloseParenToken": Any(),
+				"IsMissing": Bool(false),
+				"IsStructuredTrivia": Bool(false),
+				"Parameters": Var("params"),
+			}},
+			{Name: "ReturnType", Optional: "optReturn", Op: Cases("cases_return",
+				// Type = PredefinedType
+				Obj{
+					uast.KeyType: String("PredefinedType"),
+					uast.KeyPos: Any(),
+					"IsMissing": Bool(false),
+					"IsStructuredTrivia": Bool(false),
+					"IsUnmanaged": Any(),
+					"IsVar": Any(),
+					"Keyword": Obj{
+						uast.KeyType: Any(),
+						uast.KeyPos: Any(),
+						"IsMissing": Bool(false),
+						"LeadingTrivia": Any(),
+						"TrailingTrivia": Any(),
+						"Text": Var("rettype_text"),
+						"Value": Var("rettype_text"),
+						"ValueText": Var("rettype_text"),
+					},
+				},
+				// Type = Identifier (User type)
+				Var("rettype_ident"),
```

Comment:
This is dangerous - it matches anything. `Check(HasType(), Var())` would be better. But with the comment below `Cases` won't be needed.

Reply:
Err, it's not below anymore :grin:  This one: https://github.com/bblfsh/csharp-driver/pull/17#discussion_r

Reply:
Can't see the comment, link goes to the top of this PR.

Reply:
Second look: since the `Argument.Type` field is `Any`, shouldn't we just get the native `ReturnType` var and be done with the switch? I'll do this on the next commit, PTAL.